### PR TITLE
[*] Technical - Fix linted deleted files issue

### PR DESCRIPTION
**Pull request by slimee**
This PR exclude from pre-commit-hook linter files that will not be commited. This includes deleted files, untracked or not staged for commit changed files.

### CODE REVIEW

File: .eslint-bin/pre-commit-hook.js [hunk 1]
```diff
@@ -3,6 +3,12 @@ const simpleGit = require('simple-git')(`${__dirname}/..`);
 
 let listFilesModified = [];
 
+function excludeNonCommitedFiles(file) {
+  return file.index !== 'D' // deleted
+  && file.index !== ' ' // not staged for commit
+  && file.index !== '?'; // untracked
```
Comment (VincentMolinie) [inline .eslint-bin/pre-commit-hook.js]: [Code Style] Missing indent

Comment (VincentMolinie) [inline .eslint-bin/pre-commit-hook.js]: [Code Convention] All comment prefixed by NOTICE, FIXME or TODO

File: .eslint-bin/pre-commit-hook.js [hunk 2]
```diff
@@ -3,6 +3,12 @@ const simpleGit = require('simple-git')(`${__dirname}/..`);
 
 let listFilesModified = [];
 
+function excludeNonCommitedFiles(file) {
```
Comment (VincentMolinie) [inline .eslint-bin/pre-commit-hook.js]: Why ? 🤔

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,9 @@
 - Technical - Upgrade ESLint rules.
 - Technical - Ensure that all files follow the ESLint rules.
 
+### Fixed
+- Technical - Exclude from pre-commit-hook linter files that are not commited.
```
Comment (VincentMolinie) [inline CHANGELOG.md]:
```suggestion
- Technical - Exclude from pre-commit-hook linter files that are not commit.
```

File: .eslint-bin/pre-commit-hook.js (hunk 2 above)
Comment (slimee) [reply .eslint-bin/pre-commit-hook.js]: The user working copy belongs to the userland.
`pre-commit-hook` job is to control files that the user is going to send into the commit, not everything present in the working copy.

File: .eslint-bin/pre-commit-hook.js [hunk 4]
```diff
@@ -3,6 +3,12 @@ const simpleGit = require('simple-git')(`${__dirname}/..`);
...
+  return file.index !== 'D' // NOTICE deleted files
+    && file.index !== ' ' // NOTICE files not staged for commit
+    && file.index !== '?'; // NOTICE untracked files
```
Comment (VincentMolinie) [inline .eslint-bin/pre-commit-hook.js]: Sorry forgot to tell you `NOTICE: Deleted files`...

Comment (VincentMolinie) [reply .eslint-bin/pre-commit-hook.js]: One semi colon and the first letter of your comment in capital letter

File: CHANGELOG.md (hunk 3 above)
Comment (VincentMolinie) [inline CHANGELOG.md]: committed

**Pull request merged**